PhysicsTools/Heppy/python/analyzers/objects/JetAnalyzer.py

```diff
@@ -238,10 +239,12 @@ def process(self, event):
             else:
                 photons = [ g for g in event.selectedPhotons ] 
 
-        self.gamma_cleanJetsAll = cleanNearestJetOnly(self.cleanJetsAll, photons, self.jetGammaDR)
+#        self.gamma_cleanJetsAll = cleanNearestJetOnly(self.cleanJetsAll, photons, self.jetGammaDR)
+        self.gamma_cleanJetsAll = cleanNearestJetOnly(jetsEtaCut, photons+leptons, self.jetGammaDR)
         self.gamma_cleanJets    = [j for j in self.gamma_cleanJetsAll if abs(j.eta()) <  self.cfg_ana.jetEtaCentral ]
         self.gamma_cleanJetsFwd = [j for j in self.gamma_cleanJetsAll if abs(j.eta()) >= self.cfg_ana.jetEtaCentral ]
-        self.gamma_noIdCleanJetsAll = cleanNearestJetOnly(self.noIdCleanJetsAll, photons, self.jetGammaDR)
+#        self.gamma_noIdCleanJetsAll = cleanNearestJetOnly(self.noIdCleanJetsAll, photons, self.jetGammaDR)
+        self.gamma_noIdCleanJetsAll = cleanNearestJetOnly(self.jetsAllNoID, photons+leptons, self.jetGammaDR)
```

Comment:
I did use the jet cleaned from the photons for a while in H->ZZ for example, and it may be used e.g. by the mono-photon people (the pool of heppy users is much larger than just the 3-4 susy analyses).

The new commit is basically a rename of the `jetGammaDR` to `jetGammaLepDR` since nothing else in the code afterwards uses `jetGammaDR`, so it doesn't really change anything besides leaving an extra configurable parameter that does nothing, so I'd drop it.


Two possible solutions:
 * add a switch `cleanFromLepAndGammaSimultaneously`: if true, use the logic of this PR and raise an exception at configuration time if the two deltaR's are different; if false, stay with the old (first clean from the leptons with the lepton DR, then clean from the jets)
 * make an extended cleanNearestJetOnly that it can use different deltaR's for the two classes of objects. 

The first option seems preferable to me